### PR TITLE
[KOA-4464] migrate more examples to new story format

### PR DESCRIPTION
**Pull request by k0nserv**
Migrated stories:

- Dialog
- Horizontal Nav
- Labels

Todo:

- [ ] Retake screenshots(for verification)

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Example/Backpack Screenshot/Screenshots.swift [hunk 1]
```diff
@@ -282,7 +282,7 @@ class Screenshots: XCTestCase {
         app.tables.staticTexts["Default"].tap()
         saveScreenshot(component: "label", scenario: "default", userInterfaceStyle: userInterfaceStyle)
         tapBackButton()
-        app.tables.staticTexts["Emphasized"].tap()
+        app.tables.staticTexts["Emphasised"].tap()
```
Comment (shaundon) [inline Example/Backpack Screenshot/Screenshots.swift:285]: ♥️

**Review by shaundon — APPROVED** [review]
Love to see a diff that deletes so much code 🎉

File: Example/Backpack/Stories/LabelStory.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
+//
```
Comment (georgegillams) [inline Example/Backpack/Stories/LabelStory.swift:1]: extraneous comment

File: Example/Backpack/Stories/DialogStory.swift [hunk 3]
```diff
@@ -0,0 +1,117 @@
+//
```
Comment (georgegillams) [inline Example/Backpack/Stories/DialogStory.swift:1]: extraneous comment

File: Example/Backpack/Stories/HorizontalNavigationStory.swift [hunk 4]
```diff
@@ -0,0 +1,83 @@
+//
```
Comment (georgegillams) [inline Example/Backpack/Stories/HorizontalNavigationStory.swift:1]: extraneous comment

File: Example/Backpack/Stories/HorizontalNavigationStory.swift [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    var presentableStory: Presentable {
+        let storyboard = loadStoryboard(name: "Main")
+        let viewController = storyboard("HorizontalNavViewController").makeViewController
+
+        return CustomPresentable(generateViewController: enrich(viewController, self.configureStory))
+    }
+
+    // This isn't actually complex, it's just a large switch
```
Comment (georgegillams) [inline Example/Backpack/Stories/HorizontalNavigationStory.swift:46]: Having a large number of different paths through software is the definition of cyclomatic complexity. I think in this case `M=13`.
So this comment is a lie.

**Review by georgegillams — APPROVED** [review]
Only a couple of nitpicks from me. Looks good 👍

File: Example/Backpack/Stories/HorizontalNavigationStory.swift (hunk 5 above)
Comment (k0nserv) [reply Example/Backpack/Stories/HorizontalNavigationStory.swift:46]: Yes, but cyclomatic complexity is a bit flawed in this way it tries to objectively measure complexity but I think this method subjectively isn't complex. Any change I could make to this function would not make it less complex or easier to read

File: Example/Backpack/Stories/HorizontalNavigationStory.swift (hunk 4 above)
Comment (k0nserv) [reply Example/Backpack/Stories/HorizontalNavigationStory.swift:1]: I think these might be from the project template

Comment (k0nserv) [reply Example/Backpack/Stories/HorizontalNavigationStory.swift:1]: Or is it that limitation in Xcode?

Comment (georgegillams) [reply Example/Backpack/Stories/HorizontalNavigationStory.swift:1]: I'm not sure - I just know I try to kill them off whenever I see them

File: Example/Backpack Screenshot/Screenshots.swift [hunk 6]
```diff
@@ -24,11 +24,17 @@ import XCTest
...
 class Screenshots: XCTestCase {
 
-    var app: XCUIApplication = XCUIApplication()
+    var app: XCUIApplication!
 
-    override func setUp() {
-        // Put setup code here. This method is called before the invocation of each test method in the class.
+    func createApp() -> XCUIApplication {
+        let app = XCUIApplication()
+        app.launchArguments.append("UITests")
```
Comment (k0nserv) [inline Example/Backpack Screenshot/Screenshots.swift:31]: This makes the screenshots much faster by disabling animations 👍🏼

**Review by georgegillams — APPROVED** [review]

**Pull request merged**